### ISSUE TITLE
scikit flow - race condition when trying to fit

### ISSUE THREAD

**Issue by mikecassell** [tensorflow/tensorflow#3605]
I'm probably missing something obvious but I did a fresh build tonight and when trying to run the MNIST examples, every time I run them I'm getting the below errors. I have tried a few of the examples and can't seem to get around it (even when updating the depreciated bits still in the examples.) Am I missing something?

Code:
`
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

from sklearn import metrics
import tensorflow as tf
from tensorflow.contrib import learn
import numpy as np

### Download and load MNIST data.

mnist = learn.datasets.load_dataset('mnist')
y = np.array(mnist.train.labels, dtype='int32')
### Linear classifier.

feature_columns = learn.infer_real_valued_columns_from_input(mnist.train.images)
classifier = learn.TensorFlowLinearClassifier(
    feature_columns=feature_columns, n_classes=10, batch_size=100, steps=1000,
    learning_rate=0.01)
classifier.fit(mnist.train.images,y)
score = metrics.accuracy_score(
    mnist.test.labels, classifier.predict(mnist.test.images))
print('Accuracy: {0:f}'.format(score))
`

Error:
I tensorflow/core/common_runtime/gpu/gpu_device.cc:839] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GT 750M, pci bus id: 0000:07:00.0)
E tensorflow/c/c_api.cc:485] WhereOp: Race condition between counting the number of true elements and writing them.  When counting, saw 1287 elements; but when writing their indices, saw 7 elements.
	 [[Node: report_uninitialized_variables/Where = Where[_device="/job:localhost/replica:0/task:0/cpu:0"](report_uninitialized_variables/Reshape_1)]]
	 [[Node: report_uninitialized_variables/Where/_28 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_30_report_uninitialized_variables/Where", tensor_type=DT_INT64, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

**Comment by mikecassell** [comment]
I tried running it as a script (instead of in Jupyter) and it runs fine, is there a known issue with running in notebooks?
